### PR TITLE
Core/Battleground: Clean up some Team/TeamId parameters to use enums …

### PR DESCRIPTION
**Pull request by Golrag**
…instead of integer types

**Changes proposed:**

-  Change some integer types to Enum types in parameters/return types/members

**Issues addressed:**

Closes: None

**Tests performed:**

- Builds

**Known issues and TODO list:**

- None

### CODE REVIEW

File: src/server/game/World/World.h [hunk 1]
```diff
@@ -645,10 +645,10 @@ class TC_GAME_API World
...
         void SendGMText(uint32 string_id, ...);
         void SendServerMessage(ServerMessageType messageID, std::string_view stringParam = {}, Player const* player = nullptr);
-        void SendGlobalMessage(WorldPacket const* packet, WorldSession* self = nullptr, uint32 team = 0);
-        void SendGlobalGMMessage(WorldPacket const* packet, WorldSession* self = nullptr, uint32 team = 0);
-        bool SendZoneMessage(uint32 zone, WorldPacket const* packet, WorldSession* self = nullptr, uint32 team = 0);
-        void SendZoneText(uint32 zone, const char *text, WorldSession* self = nullptr, uint32 team = 0);
+        void SendGlobalMessage(WorldPacket const* packet, WorldSession* self = nullptr, Team team = TEAM_OTHER);
+        void SendGlobalGMMessage(WorldPacket const* packet, WorldSession* self = nullptr, Team team = TEAM_OTHER);
+        bool SendZoneMessage(uint32 zone, WorldPacket const* packet, WorldSession* self = nullptr, Team team = TEAM_OTHER);
+        void SendZoneText(uint32 zone, const char *text, WorldSession* self = nullptr, Team team = TEAM_OTHER);
```
Comment (Shauren) [inline src/server/game/World/World.h:651]: These would have been better as `Optional<Team> team = {}`

**Review by Shauren — CHANGES_REQUESTED** [review]

File: src/server/game/World/World.h (hunk 1 above)
Comment (Golrag) [reply src/server/game/World/World.h:651]: Alright!

**Comment by jackpoz** [comment]
> **Tests performed:**
> 
> * Builds

Marking this PR as draft, please test it and then publish it again

**Comment by Golrag** [comment]
Tests performed (after rebase):
- Outdoor pvp Hellfire: checked if monster credit was granted
- Outdoor pvp Nagrand: halaa take over
- Outdoor pvp terrokar: tower take over quest credit
- battleground queue & win

**Review by Shauren — APPROVED** [review]

**Pull request merged**